cmd/ui-errors.go

```diff
@@ -0,0 +1,169 @@
+		"Invalid cache excludes value",
+		"Please check the passed value",
+		"MINIO_CACHE_EXCLUDE: List of cache exclusion patterns delimited by `;`.",
+	)
+
+	uiErrInvalidCacheExpiryValue = newUIErrFn(
+		"Invalid cache expiry value",
+		"Please check the passed value",
+		"MINIO_CACHE_EXPIRY: Cache expiry duration in days.",
+	)
+
+	uiErrInvalidCredentials = newUIErrFn(
+		"Passed credentials are not suitable for use",
+		"Please provide correct credentials",
+		`- Access key length should be between 3 and 20 characters.
+- Secret key should not be between 8 and 40 characters.`,
+	)
+
+	uiErrInvalidErasureEndpoints = newUIErrFn(
+		"Endpoints unsuitable to activate Erasure mode",
+		"Please provide correct combinations of local/remote paths",
+		"",
+	)
+
+	uiErrInvalidNumberOfErasureEndpoints = newUIErrFn(
+		"Number of endpoints are not suitable for use in Erasure mode",
+		"Please provide an even number of endpoints greater or equal to 4",
+		"",
+	)
+
+	uiErrStorageClassValue = newUIErrFn(
+		"Invalid storage class value",
+		"Please check the passed value",
+		`MINIO_STORAGE_CLASS_STANDARD: Format "EC:<Default_Parity_Standard_Class>" (e.g. "EC:3"). This sets the number of parity disks for Minio server in Standard mode. Object are stored in Standard mode, if storage class is not defined in Put request.
+MINIO_STORAGE_CLASS_RRS: Format "EC:<Default_Parity_Reduced_Redundancy_Class>" (e.g. "EC:3"). This sets the number of parity disks for Minio server in reduced redundancy mode. Objects are stored in Reduced Redundancy mode, if Put request specifies RRS storage class.`,
+	)
+
+	uiErrUnexpectedBackendVersion = newUIErrFn(
+		"Backend version seems to be too recent",
+		"Please update to the latest Minio version",
+		"",
+	)
+
+	uiErrInvalidAddressFlag = newUIErrFn(
+		"--address parameter is not suitable for use",
+		"Please check --address parameter",
+		`--address binds a specific ADDRESS:PORT, ADDRESS can be an  IP or hostname (default:':9000')
+    Examples: --address ':443'
+              --address '172.16.34.31:9000'`,
+	)
+
+	uiErrInvalidFSEndpoint = newUIErrFn(
+		"The given endpoint is not suitable to activate standalone mode",
+		"Please check the given FS endpoint",
+		`Standalone mode requires only one local disk path to be activated.
+Example 1:
+   $ minio server /var/data/minio/`,
+	)
+
+	uiErrUnableToWriteInBackend = newUIErrFn(
+		"Unable to write on the specified endpoint",
+		"Please ensure that Minio binary has enough permissions to write in the backend",
+		"",
+	)
+
+	uiErrUnableToReadFromBackend = newUIErrFn(
+		"Unable to read data from the backend",
+		"Please ensure that Minio binary has enough permission to read data from the backend",
+		"",
+	)
+
+	uiErrPortAlreadyInUse = newUIErrFn(
+		"Port is already in use",
+		"Please ensure no other program is reserving the same address/port",
+		"",
+	)
+
+	uiErrNoPrivilegesToAccessDirFiles = newUIErrFn(
+		"Missing privileges to access to the specified path",
```

Comment:
`privileges` is also confusing we should simply write the lower level error here. Because of *os.PathError happens for a lot of different reasons other than `privileges` only.

Reply:
The problem is with the word privileges. OS package uses os.IsPermission() not os.IsPrivileged() it gives an entirely different meaning in an operating system context.

Reply:
> The problem is with the word privileges. OS package uses os.IsPermission() not os.IsPrivileged() it gives an entirely different meaning in an operating system context.

oh okay, will reword all "privileges" to permissions